main.py

```diff
@@ -183,167 +190,122 @@ def compose_message(issues):
-            async for issue in issues:
-                chosen_issues.append(issue)
-                if len(chosen_issues) == number_of_issues:
-                    break
-        else:
-            title_suffix = " in review queue:"
-            chosen_issues = []
-            async for issue in issues:
-                chosen_issues.append(issue)
-                if len(chosen_issues) == number_of_issues:
-                    break
-
-        reply.title = f"Oldest {number_of_issues} PR" + title_suffix
-        reply.description = compose_message(beautify_issues(chosen_issues))
-        if len(chosen_issues) < number_of_issues:
-            reply.set_footer(text="There weren't enough...")
-    await channel.send(embed=reply)
-    await set_playing('The Waiting Game')
+    await bot.change_presence(activity=discord.Game(name=status))
 
 
-@client.event
+@bot.event
 async def on_ready():
     await set_playing('The Waiting Game')
 
 
-@client.event
-async def on_message(message):
-    # we do not want the bot to reply to itself
-    if message.author.id == client.user.id:
-        return
-    # Don't reply to bots
-    if message.author.bot:
-        return
-    channel = message.channel
-    data = message.content
-    if len(data) == 0 or data[0] != '!':
-        return
-    # split message into command and arguments
-    query = data.strip().split(' ')
-    command = query[0][1:]
-    args = query[1:]
+async def modify_params(ctx, conf, name, number_of_issues, noun=None, channel=None, error_channel=default_error_handler):
+    author=None
+
+    if number_of_issues < 1:
+        await error_channel("Woah, there!", "I can't give you un-natural issues." + \
+            " I'm not a monster!!")
+    if number_of_issues > 10:
+        number_of_issues = 10
+        await error_channel("Woah, there!", "Let's curb that enthusiasm.. just a little")
+    
+    if name=="rq":
+        if noun in ["issue", "issues", "ISSUE", "ISSUES"]:
+            await error_channel("Woah, there!", "Sorry, but we don't review issues here. Up for some PRs?")
+        if noun is None or noun not in ["all", "ALL"]:
+            author = ctx.message.author
+            if isinstance(author, discord.Member):
+                author = author.nick or author.name
+            elif isinstance(author, discord.User):
+                author = author.name
+    else:
+        if noun in ["issue", "issues", "ISSUE", "ISSUES"]:
+            conf["pull_request"]=False
+        elif noun in ["pr", "prs", "PR", "PRs", "PRS"]:
+            conf["pull_request"]=True
+        else:
+            conf["pull_request"]=False
+            await error_channel("", "Insufficient info, defaulting to issues..")
+
+    return number_of_issues, conf, author
+
+
+for name, conf in command_config.items():
+    @bot.command(name=name)
+    async def magic_function(ctx, number_of_issues: int, noun=None, channel=None, name=name, conf=conf): # name=name and conf=conf used to prevent late binding
```

Comment:
You do a better name than `magic_function` 😆

Reply:
:disappointed: